pkg/resmgr/control/cpu/cpu.go

```diff
@@ -271,28 +265,26 @@ func (ctl *cpuctl) configure(cfg *cfgapi.Config) error {
 
 	// Configure the system
 	for class, cpus := range assignments {
-		if _, ok := ctl.classes[class]; ok {
-			// Re-configure cpus (sysfs) according to new class parameters
-			if err := ctl.enforceCpufreq(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-			if err := ctl.enforceCpufreqGovernor(class, cpus.SortedMembers()...); err != nil {
-				log.Error("cpufreq enforcement on re-configure failed: %v", err)
-			}
-		} else {
+		if _, ok := ctl.classes[class]; !ok {
 			// TODO: what should we really do with classes that do not exist in
 			// the configuration anymore? Now we remember the CPUs assigned to
 			// them. A further config update might re-introduce the class in
 			// which case the CPUs will be reconfigured.
-			log.Warn("class %q with cpus %v missing from the configuration", class, cpus)
+			return fmt.Errorf("non-existent cpu class %q", class)
+		}
+		// Re-configure cpus (sysfs) according to new class parameters
+		if err := ctl.enforceCpufreq(class, cpus.SortedMembers()...); err != nil {
+			log.Error("cpufreq enforcement on re-configure failed: %v", err)
+		}
+		if err := ctl.enforceCpufreqGovernor(class, cpus.SortedMembers()...); err != nil {
+			log.Error("cpufreq governor enforcement on re-configure failed: %v", err)
 		}
 	}
 	if err := ctl.enforceUncore(assignments); err != nil {
 		log.Error("uncore frequency enforcement on re-configure failed: %v", err)
 	}
 
 	log.Debug("cpu controller configured")
-
```

Comment:
nit: unrelated white space change in a code block not touched by the PR.

Reply:
fixed